### ISSUE TITLE
gresetinitiallines{2} broken?

### ISSUE THREAD

**Issue by eroux** [gregorio-project/gregorio#1026]
Compiling this document:

```tex
\documentclass[11pt]{article} % use larger type; default would be 10pt
\usepackage{fontspec}
\usepackage{graphicx} % support the \includegraphics command and options
\usepackage{geometry} % See geometry.pdf to learn the layout options. There are lots.
\geometry{a5paper} % or letterpaper (US) or a5paper or....
\usepackage[]{gregoriotex} % for gregorio score inclusion
\usepackage{libertine} % Decent (free) font for Gregorian, but should be changed if you have high standards

\gresetinitiallines{2}

\begin{document}

\gregorioscore[a]{PopulusSion}

\end{document}
```

with the PopulusSion.gabc from `examples/`, I get this weird warning:

```
Package GregorioTeX Warning: biginitial style is deprecated, but the old
(GregorioTeX)                behavior will be used since deprecated usage is
(GregorioTeX)                currently enabled. This behavior will disappear
(GregorioTeX)                in 5.0, so consider disabling deprecated usage
(GregorioTeX)                and switching to the initial style instead. (See
(GregorioTeX)                UPGRADE.md for more information) on input line 19.
```

and this wrong output (even after several compilations): 

![buginitial](https://cloud.githubusercontent.com/assets/60868/13629914/70dfbf3c-e5dc-11e5-940b-dd299da6b934.png)

I there something I'm doing wrong?

**Comment by eroux** [comment]
Works when forcing the first two end of lines. I think it's mainly a documentation problem. Also, such an example could appear in Tips and tricks?

**Comment by henryso** [comment]
I use this trick for a three-line initial.  I manually break the first and third lines and let LaTeX break the middle one.  It might not be a good idea to encourage this trick, as things will likely break when we put in proper support for more-than-two-line initials.

**Comment by eroux** [comment]
Right... maybe the most correct way would be to have a `z1` (or anything) in gabc to tell GregorioTeX "the last line of the part after the initial ends here" ? I'm quite sure we could play with `\parshape`, maybe it would solve the problem in an elegant way...

**Comment by eroux** [comment]
Well, it works good enough for 4.1.1, let's not block it with that...

**Comment by rpspringuel** [comment]
Where in the documentation should this be added/improved?

✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝✝
Br. Samuel, OSB
(R. Padraic Springuel)
PAX ☧ ΧΡΙΣΤΟΣ

**Comment by eroux** [comment]
Well, documenting it in `\gresetinitiallines` is certainly a good start, though I doubt many people will look there if they encounter the same problem as I did... I think a check in `\GreEndScore` would be best: if `gre@biginitial = 1` and `gre@knownline < 3`, then the user didn't input the forced end of line

**Comment by rpspringuel** [comment]
I at least partially addressed this in 4.1.1.  Is what I did sufficient or is there something else which remains to be done?

**Comment by eroux** [comment]
In which commit/PR was it addressed?

**Comment by rpspringuel** [comment]
cd714a6

**Comment by eroux** [comment]
ok, perfect then!

**Comment by eroux** [comment]
Sorry, reopening as I just got a complain about the deprecation message, which is not very helpful... Maybe it should be rephrased to say that `biginitialstyle` is currently used, but in order to prepare for next release, `initialstyle` should be set to the same value for big initials. What do you think?

**Comment by henryso** [comment]
The message is currently
```
biginitial style is deprecated, but the old behavior will be used since deprecated usage is currently enabled. This behavior will disappear  in 5.0, so consider disabling deprecated usage and switching to the initial style instead. (See UPGRADE.md for more information)
``

How about
```
biginitial style is deprecated, but since you are allowing deprecated usage, Gregorio will continue to use your biginitial style as it did before.  However, in upcoming version 5.0, Gregorio will only support one initial style for both big and small initials.  Consider disabling deprecated usage, which will then require you to set the initial style before the appropriate scores to style all initials: set the initial style for a big initial before a score with a big initial and back for a small initial afterwards.
```

Too wordy?  To confusing?  Any ideas to fix it?

**Comment by eroux** [comment]
Looks good! Maybe it could help the user by giving the command not to allow deprecated commands?

**Comment by henryso** [comment]
If I do that, it would look like:

`biginitial style is deprecated, but since you are allowing deprecated usage, Gregorio will continue to use your biginitial style as it did before. However, in upcoming version 5.0, Gregorio will only support one initial style for both big and small initials. Consider disabling deprecated usage by passing the [allowdeprecated=false] package option, which will then require you to set the initial style before the appropriate scores to style all initials: set the initial style for a big initial before a score with a big initial and back for a small initial afterwards. (See UPGRADE.md for more information)`

However, I do have a problem with this.  The instruction to add `[allowdeprecated=false]` is only true for LaTeX.  For plain TeX, people would need to do something like:
```tex
\edef\greoldcatcode{\the\catcode`@}%
\catcode`\@=11%
\gre@allowdeprecatedfalse%
\catcode`\@=\greoldcatcode%
```

Should we put in the extra text only if LaTeX is detected?

**Comment by eroux** [comment]
I think just taking LaTeX into account is just fine, people using Plain TeX are full of resources and will be able to figure it out. So just the text of your last comment is fine!
